lib/procedures/ensure-manta-deployment-gz-links.js

```diff
@@ -0,0 +1,185 @@
+                               ctx.currServerUuid) {
+                        next(new VError(
+                            '"manta" instance %s (%s) is not on this server: ' +
+                                'server_uuid=%s, currServerUuid=%s',
+                            insts[0].uuid,
+                            insts[0].params.alias,
+                            insts[0].params.server_uuid,
+                            ctx.currServerUuid));
+                    } else {
+                        ctx.inst = insts[0];
+                        next();
+                    }
+                });
+            },
+
+            // Create stubs in /opt/smartdc/bin/ to a few `manta-*` tools
+            // in the manta-deployment zone.
+            function createLinks(ctx, next) {
+                let zoneBaseDir = '/zones/' + ctx.inst.uuid +
+                    '/root/opt/smartdc/manta-deployment';
+
+                fs.writeFileSync(
+                    '/opt/smartdc/bin/manta-login',
+                    [
+                        '#!/bin/bash',
+                        'exec ' + zoneBaseDir + '/bin/manta-login "$@"'
+                    ].join('\n'),
+                    {
+                        mode: 0o755
+                    }
+                );
+                fs.writeFileSync(
+                    '/opt/smartdc/bin/manta-adm',
+                    [
+                        '#!/bin/bash',
+                        'exec ' + zoneBaseDir + '/build/node/bin/node ' +
+                            zoneBaseDir + '/bin/manta-adm "$@"'
+                    ].join('\n'),
+                    {
+                        mode: 0o755
+                    }
+                );
+                fs.writeFileSync(
+                    '/opt/smartdc/bin/manta-oneach',
+                    [
+                        '#!/bin/bash',
+                        'exec ' + zoneBaseDir + '/build/node/bin/node ' +
+                            zoneBaseDir + '/bin/manta-oneach "$@"'
+                    ].join('\n'),
+                    {
+                        mode: 0o755
+                    }
+                );
+                ui.info(
+                    'Wrote "/opt/smartdc/bin/manta-{login,adm,oneach}" stubs');
+
+                let manpages;
+                let mandir = zoneBaseDir + '/man/man1';
+                try {
+                    manpages = fs.readdirSync(mandir);
+                } catch (readdirErr) {
+                    next(new VError(readdirErr,
+                        'could not read manta-deployment man dir'));
+                    return;
+                }
+                for (let manpage of manpages) {
+                    let link = '/opt/smartdc/man/man1/' + manpage;
+                    let targ = mandir + '/' + manpage;
+                    try {
+                        fs.lstatSync(link);
+                        fs.unlinkSync(link);
+                    } catch (_lstatErr) {
```

Comment:
nit: Is this really only an _lstatErr? Could it not also be an unlink error? E.g. What if someone mistakenly created a directory with the same name as `link` here.

Reply:
Updated to skip if the symlink path already exists (as setup_manta_zone.sh does).